Sources/TuistKit/GraphMappers/AutomationGraphMapperProvider.swift

```diff
@@ -22,6 +22,7 @@ final class AutomationGraphMapperProvider: GraphMapperProviding {
         mappers.append(
             TestsCacheGraphMapper(hashesCacheDirectory: testsCacheDirectory, config: config)
         )
+        mappers.append(CacheTreeShakingGraphMapper())
```

Comment:
Whats this cange for?

Reply:
This mapper is getting rid of the targets marked as `prune`.

Reply:
As I mentioned in one of the comments, I'd extract the logic of ensuring the project integrity into a new mapper and use that one instead.